client/client.go

```diff
@@ -221,6 +222,12 @@ func (client *Client) raw(method, urlpath string, query url.Values, headers map[
 		req.Header.Set(AllowInteractionHeader, "true")
 	}
 
+	if timeout > 0 {
+		ctx, cancel := context.WithTimeout(req.Context(), timeout)
```

Comment:
Yay for using `context.WithTimeout()`

At the same, IIRC the timeout is for the complete request. Ones that take log (logs?) could fail in some unexpected ways.